### PR TITLE
Pr persistent partition and mount disk

### PR DESCRIPTION
**Pull request by kaleo211**

### CODE REVIEW

File: agent/action/mount_disk.go [hunk 1]
```diff
@@ -59,11 +64,23 @@ func (a MountDiskAction) Run(diskCid string) (interface{}, error) {
...
+	}
+
+	devicePath, isMountPoint, err := a.diskMounter.IsMountPoint(mountPoint)
 	if err != nil {
 		return nil, bosherr.WrapError(err, "Checking mount point")
 	}
+	a.logger.Info(logTag, "realPath = %s, devicePath = %s, isMountPoint = %s", realPath, devicePath, isMountPoint)
+
 	if isMountPoint {
+		if realPath == devicePath || fmt.Sprint(realPath+"1") == devicePath {
```
Comment (mariash) [inline agent/action/mount_disk.go]: @kaleo211 what is the purpose of comparing devicePath with realPath+1? It is unclear from code and there are no tests that are failing when removing this condition.

Comment (kaleo211) [inline agent/action/mount_disk.go]: in our case, we have a fixed persistent disk, and it always be /dev/sdb. And bosh agent always mounts /dev/sdb1 to /var/vcap/store for the first time. when we want to remount, we want to know sdb is already mounted by that line.

Comment (mariash) [inline agent/action/mount_disk.go]: If bosh-agent always mount it as `/dev/sdb1` why there is a first condition?

Comment (kaleo211) [inline agent/action/mount_disk.go]: we keep that for other cpi, I don't know whether it make sense, :))

Comment (mariash) [inline agent/action/mount_disk.go]: Partitioning is related to OS and should not depend on which CPI it is running on. So it is unclear how it should be affected by other cpi...

Comment (kaleo211) [inline agent/action/mount_disk.go]: I see, we will remove that condition and rerun the integration test,

**Comment by mariash** [comment]
Hi @kaleo211,

We are seeing test failure: `platform/dummy_platform_test.go:97`. We are going to fix it as part of merge. Just want to let you know so that you can improve your tests run in future.

Thanks.

**Pull request closed without merging**